hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -158,4 +163,36 @@ public static boolean sumHadOverflow(long a, long b, long sum) {
     public static boolean diffHadOverflow(long a, long b, long diff) {
         return ((a ^ b) & (a ^ diff)) < 0;
     }
+
+    /**
+     * Checks that the {@code object} implements {@link Serializable} and is
+     * correctly serializable by actually trying to serialize it. This will
+     * reveal some non-serializable field early.
+     *
+     * @param object object to check
+     * @param objectName object description for the exception
+     * @throws IllegalArgumentException if {@code object} is not serializable
+     */
+    public static void checkSerializable(Object object, String objectName) {
+        if (object != null) {
+            if (!(object instanceof Serializable)) {
+                throw new IllegalArgumentException("\"" + objectName + "\" must be serializable");
+            }
+            try  (ObjectOutputStream os  = new ObjectOutputStream(new NullOutputStream())) {
+                os.writeObject(object);
+            } catch (NotSerializableException | InvalidClassException e) {
+                throw new IllegalArgumentException("\"" + objectName + "\" must be serializable", e);
+            } catch (IOException e) {
+                // never really thrown, as the underlying stream never throws it
```

Comment:
This comment is incorrect because `ObjectOutputStream` may throw `IOException` itself. Given this fact, it would be better to wrap in `JetException` instead of the generic `RuntimeException`.